src/NuGet.Indexing/NuGetSearcherManager.cs

```diff
@@ -8,74 +8,80 @@
-using Lucene.Net.Store.Azure;
-using Microsoft.WindowsAzure.Storage;
 using Microsoft.Extensions.Logging;
 using Newtonsoft.Json;
 using FrameworkLogger = Microsoft.Extensions.Logging.ILogger;
+using System.Threading.Tasks;
+using NuGet.Services.Configuration;
+using Directory = Lucene.Net.Store.Directory;
 
 namespace NuGet.Indexing
 {
     public class NuGetSearcherManager : SearcherManager<NuGetIndexSearcher>
     {
         public static readonly TimeSpan AuxiliaryDataRefreshRate = TimeSpan.FromHours(1);
+        
+        private readonly FrameworkLogger _logger;
 
+        private DateTime _lastTimeIndexReloaded = DateTime.MinValue;
+        private readonly TimeSpan _indexReloadRate;
+        private readonly IIndexDirectoryProvider _indexProvider;
         private readonly ILoader _loader;
-        private readonly FrameworkLogger _logger;
-        private readonly AzureDirectorySynchronizer _azureDirectorySynchronizer;
 
-        private DateTime _auxiliaryDataReloaded = DateTime.MinValue;
+        private DateTime _lastTimeAuxiliaryDataRefreshed = DateTime.MinValue;
+        private readonly TimeSpan _auxiliaryDataRefreshRate;
         private readonly IDictionary<string, HashSet<string>> _owners = new Dictionary<string, HashSet<string>>(StringComparer.OrdinalIgnoreCase);
         private readonly IDictionary<string, HashSet<string>> _curatedFeeds = new Dictionary<string, HashSet<string>>();
         private readonly Downloads _downloads = new Downloads();
         private IReadOnlyDictionary<string, int> _rankings;
 
         private QueryBoostingContext _queryBoostingContext = QueryBoostingContext.Default;
 
-        public NuGetSearcherManager(string indexName,
-            FrameworkLogger logger,
-            Lucene.Net.Store.Directory directory,
+        public NuGetSearcherManager(FrameworkLogger logger,
+            IIndexDirectoryProvider indexProvider,
             ILoader loader,
-            AzureDirectorySynchronizer azureDirectorySynchronizer)
-            : base(directory)
+            int auxiliaryDataRefreshRateSec = IndexingSettings.AuxiliaryDataRefreshRateSecDefault,
+            int indexReloadRateSec = IndexingSettings.IndexReloadRateSecDefault)
         {
             if (logger == null)
             {
                 throw new ArgumentNullException(nameof(logger));
             }
 
             _logger = logger;
-
-            IndexName = indexName;
+            
             RegistrationBaseAddress = new Dictionary<string, Uri>();
 
+            _indexProvider = indexProvider;
             _loader = loader;
-            _azureDirectorySynchronizer = azureDirectorySynchronizer;
+
+            _indexReloadRate = TimeSpan.FromSeconds(indexReloadRateSec);
+            _auxiliaryDataRefreshRate = TimeSpan.FromSeconds(auxiliaryDataRefreshRateSec);
         }
 
-        public string IndexName { get; private set; }
+        public virtual string IndexName => _indexProvider.GetIndexContainerName();
```

Comment:
[Not really.](https://github.com/NuGet/NuGet.Services.Metadata/pull/91/files#diff-76f506fbfd233fe530713bf9da2104fdR36)

I thought this was the least code impact of the possible options since you cannot pass in a `IndexDirectoryProvider` into `Create`.